clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -680,35 +680,67 @@ namespace {
 QualType getItemType(const FieldDecl *FD) { return FD->getType(); }
 QualType getItemType(const CXXBaseSpecifier &BS) { return BS.getType(); }
 
+// These enable handler execution only when previous handlers succeed.
+template <typename T>
+static bool handleField(FieldDecl *FD, QualType FDTy, T &t) {
+  return (t.first->*t.second)(FD, FDTy);
+}
+template <typename T, typename... Tn>
+static bool handleField(FieldDecl *FD, QualType FDTy, T &t, Tn &... tn) {
+  return (t.first->*t.second)(FD, FDTy) && handleField(FD, FDTy, tn...);
```

Comment:
I think this "&&" logic needs to happen at every level here.  All the way to the root fields, base classes, etc.  As soon as we discover an error in attempting anything we need to give up.  So the 'bool' needs to be propogated quite a bit more.

I would think it should cause us to stop descending as well.